internal/literals/numbers.go

```diff
@@ -0,0 +1,326 @@
+		call = obfuscateInt(intValue)
+	case types.Typ[types.Int8]:
+		call = obfuscateInt8(int8(intValue))
+	case types.Typ[types.Int16]:
+		call = obfuscateInt16(int16(intValue))
+	case types.Typ[types.Int32]:
+		call = obfuscateInt32(int32(intValue))
+	case types.Typ[types.Int64]:
+		call = obfuscateInt64(int64(intValue))
+	case types.Typ[types.Uint]:
+		call = obfuscateUint(uint(intValue))
+	case types.Typ[types.Uint8]:
+		call = obfuscateUint8(uint8(intValue))
+	case types.Typ[types.Uint16]:
+		call = obfuscateUint16(uint16(intValue))
+	case types.Typ[types.Uint32]:
+		call = obfuscateUint32(uint32(intValue))
+	case types.Typ[types.Uint64]:
+		call = obfuscateUint64(uint64(intValue))
+	case types.Typ[types.Uintptr]:
+		call = obfuscateUintptr(uintptr(intValue))
+	}
+
+	if call == nil {
+		return errors.New("Node is not Integer")
+	}
+
+	cursor.Replace(call)
+	return nil
+}
+
+func obfuscateUint8(data uint8) *ast.CallExpr {
+	obfuscator := randObfuscator()
+	block := obfuscator.obfuscate([]byte{byte(data)})
+	block.List = append(block.List,
+		&ast.ReturnStmt{Results: []ast.Expr{
+			convertExpr("uint8", indexExpr("data", intLiteral("0"))),
+		}},
+	)
+
+	return lambdaCall(ident("uint8"), block)
+}
+
+func obfuscateUint16(data uint16) *ast.CallExpr {
+	obfuscator := randObfuscator()
+	b := make([]byte, 2)
+	binary.LittleEndian.PutUint16(b, data)
+	block := obfuscator.obfuscate(b)
+
+	convertExpr := &ast.BinaryExpr{
+		X:  convertExpr("uint16", indexExpr("data", intLiteral("0"))),
+		Op: token.OR,
+		Y: &ast.BinaryExpr{
+			X:  convertExpr("uint16", indexExpr("data", intLiteral("1"))),
+			Op: token.SHL,
+			Y:  intLiteral("8"),
+		},
+	}
+
+	block.List = append(block.List, boundsCheckData("1"), returnStmt(convertExpr))
+
+	return lambdaCall(ident("uint16"), block)
+}
+
+func obfuscateUint32(data uint32) *ast.CallExpr {
+	obfuscator := randObfuscator()
+	b := make([]byte, 4)
+	binary.LittleEndian.PutUint32(b, data)
+	block := obfuscator.obfuscate(b)
+
+	convertExpr := &ast.BinaryExpr{
+		X: &ast.BinaryExpr{
+			X: &ast.BinaryExpr{
+				X:  convertExpr("uint32", indexExpr("data", intLiteral("0"))),
+				Op: token.OR,
+				Y: &ast.BinaryExpr{
+					X:  convertExpr("uint32", indexExpr("data", intLiteral("1"))),
+					Op: token.SHL,
+					Y:  intLiteral("8"),
+				},
+			},
+			Op: token.OR,
+			Y: &ast.BinaryExpr{
+				X:  convertExpr("uint32", indexExpr("data", intLiteral("2"))),
+				Op: token.SHL,
+				Y:  intLiteral("16"),
+			},
+		},
+		Op: token.OR,
+		Y: &ast.BinaryExpr{
+			X:  convertExpr("uint32", indexExpr("data", intLiteral("3"))),
+			Op: token.SHL,
+			Y:  intLiteral("24"),
+		},
+	}
+
+	block.List = append(block.List, boundsCheckData("3"), returnStmt(convertExpr))
+
+	return lambdaCall(ident("uint32"), block)
+}
+
+func obfuscateUint64(data uint64) *ast.CallExpr {
+	obfuscator := randObfuscator()
+	b := make([]byte, 8)
+	binary.LittleEndian.PutUint64(b, data)
+	block := obfuscator.obfuscate(b)
+	convertExpr := &ast.BinaryExpr{
```

Comment:
I sense that this could be a loop :)

Reply:
More than that; you should be able to reuse a single implementation for any integer obfuscation, because it's just a matter of "iterate for every byte in this integer and add a binary expression".